RecoLocalTracker/SiStripZeroSuppression/src/SiStripFedZeroSuppression.cc

```diff
@@ -29,99 +83,73 @@ void SiStripFedZeroSuppression::suppress(const std::vector<SiStripDigi>& in,
+    if (ladc >= lowThresh)
+      stat[i] = lowTh;
+    if (ladc >= highThresh)
+      stat[i] = highTh;
+  }
 
-    theNextFEDlowThresh = theFEDlowThresh;
-    theNext2FEDlowThresh = theFEDlowThresh;
-    thePrevFEDlowThresh = theFEDlowThresh;
-    thePrev2FEDlowThresh = theFEDlowThresh;
-    theNeighFEDlowThresh = theFEDlowThresh;
+  for (int i = 0; i < inSize; i++) {
+    auto strip = (uint32_t)in[i].strip();
+    Payload ldata;
 
-    theNextFEDhighThresh = theFEDhighThresh;
-    thePrevFEDhighThresh = theFEDhighThresh;
-    theNeighFEDhighThresh = theFEDhighThresh;
+    ldata.stat = stat[i];
+    ldata.statPrev = zeroTh;
+    ldata.statNext = zeroTh;
+    ldata.statPrev2 = zeroTh;
+    ldata.statNext2 = zeroTh;
 
     if (((strip) % 128) == 127) {
-      adcNext = 0;
-      theNextFEDlowThresh = 9999;
-      theNextFEDhighThresh = 9999;
+      ldata.statNext = zeroTh;
     } else if (i + 1 < inSize && in[i + 1].strip() == strip + 1) {
-      adcNext = in[i + 1].adc();
-      SiStripThreshold::Data thresholds_1 = threshold.getData(strip + 1, detThRange);
-      theNextFEDlowThresh =
-          static_cast<int16_t>(thresholds_1.getLth() * noise.getNoiseFast(strip + 1, detNoiseRange) + 0.5);
-      theNextFEDhighThresh =
-          static_cast<int16_t>(thresholds_1.getHth() * noise.getNoiseFast(strip + 1, detNoiseRange) + 0.5);
+      ldata.statNext = stat[i + 1];
       if (((strip) % 128) == 126) {
-        adcNext2 = 0;
-        theNext2FEDlowThresh = 9999;
+        ldata.statNext2 = zeroTh;
       } else if (i + 2 < inSize && in[i + 2].strip() == strip + 2) {
-        adcNext2 = in[i + 2].adc();
-        theNext2FEDlowThresh = static_cast<int16_t>(
-            threshold.getData(strip + 2, detThRange).getLth() * noise.getNoiseFast(strip + 2, detNoiseRange) + 0.5);
+        ldata.statNext2 = stat[i + 2];
       }
     }
 
     if (((strip) % 128) == 0) {
-      adcPrev = 0;
-      thePrevFEDlowThresh = 9999;
-      thePrevFEDhighThresh = 9999;
+      ldata.statPrev = zeroTh;
     } else if (i - 1 >= 0 && in[i - 1].strip() == strip - 1) {
-      adcPrev = in[i - 1].adc();
-      SiStripThreshold::Data thresholds_1 = threshold.getData(strip - 1, detThRange);
-      thePrevFEDlowThresh =
-          static_cast<int16_t>(thresholds_1.getLth() * noise.getNoiseFast(strip - 1, detNoiseRange) + 0.5);
-      thePrevFEDhighThresh =
-          static_cast<int16_t>(thresholds_1.getHth() * noise.getNoiseFast(strip - 1, detNoiseRange) + 0.5);
+      ldata.statPrev = stat[i - 1];
       if (((strip) % 128) == 1) {
-        adcPrev2 = 0;
-        thePrev2FEDlowThresh = 9999;
+        ldata.statPrev2 = zeroTh;
       } else if (i - 2 >= 0 && in[i - 2].strip() == strip - 2) {
-        adcPrev2 = in[i - 2].adc();
-        thePrev2FEDlowThresh = static_cast<int16_t>(
-            threshold.getData(strip - 2, detThRange).getLth() * noise.getNoiseFast(strip - 2, detNoiseRange) + 0.5);
+        ldata.statPrev2 = stat[i - 2];
       }
     }
 
```

Comment:
the old logic below was wrong in case say  ```adcNext <= adcPrev``` but _prev_ is _below_ and _next_ is _above_